frame/democracy/src/lib.rs

```diff
@@ -344,13 +238,23 @@ pub trait Trait: frame_system::Trait + Sized {
 	/// a negative-turnout-bias (default-carries) referendum.
 	type ExternalDefaultOrigin: EnsureOrigin<Self::Origin>;
 
-	/// Origin from which the next referendum proposed by the external majority may be immediately
-	/// tabled to vote asynchronously in a similar manner to the emergency origin. It remains a
-	/// majority-carries vote.
+	/// Origin from which the next majority-carries (or more permissive) referendum may be tabled to
+	/// vote according to the `FastTrackVotingPeriod` asynchronously in a similar manner to the
+	/// emergency origin. It retains its threshold method.
 	type FastTrackOrigin: EnsureOrigin<Self::Origin>;
 
-	/// Minimum voting period allowed for an fast-track/emergency referendum.
-	type EmergencyVotingPeriod: Get<Self::BlockNumber>;
+	/// Origin from which the next majority-carries (or more permissive) referendum may be tabled to
+	/// vote immediately and asynchronously in a similar manner to the emergency origin. It retains
+	/// its threshold method.
+	type InstantOrigin: EnsureOrigin<Self::Origin>;
+
+	/// Indicator for whether an emergency origin is even allowed to happen. Some chains may want
+	/// to set this permanently to `false`, others may want to condition it on things such as
+	/// an upgrade having happened recently.
+	type InstantAllowed: Get<bool>;
```

Comment:
If I understand how this will be used, it makes more sense as a Runtime Storage Item.

A new upgrade occurs, we may want an instant origin for a little while. Then once we know things are stable, we turn off the instant origin. In this case, it would be nice to turn it off without needing to do another upgrade.

Reply:
the idea is to let is be completely configurable without necessarily placing a (potentially unused) storage item that would need to be maintained separately.

e.g. it could be permanently on (like with kusama or westend) or off (like with a conservative mainnet) and it wouldn't need that storage lookup. or could be based on some other storage item (e.g. we may store the block at which the last runtime upgrade happened independently of this mechanism and it could check if it happened recently).

Reply:
I see, we just need to implement the `get()` function, which could be a storage lookup.